inc/cappuccino/UtMap.hpp

```diff
@@ -0,0 +1,468 @@
+#pragma once
+
+#include "cappuccino/Allow.hpp"
+#include "cappuccino/Lock.hpp"
+
+#include <atomic>
+#include <chrono>
+#include <deque>
+#include <optional>
+#include <unordered_map>
+#include <vector>
+#include <numeric>
+
+namespace cappuccino
+{
+/**
+ * Uniform time aware associative map for key value pairs. The associative map has no fixed size
+ * and will dynamically resize as key values are added. The map is less performant than the cache
+ * data structures, but offers unlimited size. While inserting elements the map may have poor 
```

Comment:
I don't think this is true anymore since you switched to std::map, however it would be good to let users know that any operation will run the TTL eviction policy and could be O(N) if every item in the cache happened to be expired.

Reply:
Ack